projects/ion/src/lib/pagination/pagination.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+interface Page {
+  page_number: number;
+  selected: boolean;
+}
+
+interface PageEvent {
+  actual: number;
+}
+
+export interface IonPaginationProps {
+  total: number;
+  itemsPerPage?: number;
+  size?: string;
+  events?: EventEmitter<PageEvent>;
+  allowChangeQtdItems?: boolean;
+}
+
+@Component({
+  selector: 'ion-pagination',
+  templateUrl: './pagination.component.html',
+  styleUrls: ['./pagination.component.scss'],
+})
+export class PaginationComponent implements OnInit {
+  private defaultItemsPerPage = 10;
+
+  @Input() total: IonPaginationProps['total'];
+  @Input() itemsPerPage: IonPaginationProps['itemsPerPage'] =
+    this.defaultItemsPerPage;
+  @Input() size: IonPaginationProps['size'] = 'md';
+  @Input() allowChangeQtdItems: IonPaginationProps['allowChangeQtdItems'];
+  @Output() events = new EventEmitter<PageEvent>();
+
+  public pages: Page[] = [];
```

Comment:
It would be interesting to refactor the access modifiers of Ion variables and methods in order to standardize, given that some components use public access explicitly and implicitly.